lib/puppetserver/ca/action/prune.rb

```diff
@@ -69,18 +76,25 @@ def prune_CRLs(crl_list)
               if existed_serial_number.add?(revoked.serial)
                 false
               else
+                update_flag = 1
                 @logger.debug("Removing duplicate of #{revoked.serial}, " \
                   "revoked on #{revoked.time}\n") if @logger.debug?
                 true
               end
             end
             crl.revoked=(revoked_list)
           end
+
+          return update_flag
         end
 
         def update_pruned_CRL(crl_list, pkey)
           crl_list.each do |crl|
-            crl.version=(crl.version + 1)
+            number_ext, other_ext = crl.extensions.partition{ |ext| ext.oid == "crlNumber"}
+            number_ext.each do |crl_number|
+              crl_number.value=(OpenSSL::ASN1::Integer(crl_number.value.to_i + 1))
```

Comment:
I assume this is was fixed the issue you were seeing?

Reply:
Yeah. Apparently I lose out some bytes when I used `select` as compared to `partition` which is pretty interesting.  Also, not sure if using `OpenSSL::ASN1::Integer` really helped since the `CRL#value=` try to convert the parameter to ASN1 format anywat.

Reply:
iirc, prior to this you were casting to a string via something like
```ruby
(extension.value.to_i + 1).to_s
```
maybe that was it?

Reply:
Yep. Switch back to that and lost 4 bytes and the error also came back.  Those conversions might have drop something valuable.